### ISSUE TITLE
Dll release for windows

### ISSUE THREAD

**Issue by ghostjat** [hihozhou/php-opencv#50]
Hi, 
Would you please release the dll for windows
I tried to compile it but failed
If u can it will b highly appreciated by all php windows developers

**Comment by hihozhou** [comment]
I'm trying, but I haven't succeeded yet.

**Comment by tarasfrompir** [comment]
can i already try to compile this option

**Comment by tarasfrompir** [comment]
please share dll library fow windows....

**Comment by SlardarZhang** [comment]
> I'm trying, but I haven't succeeded yet.

I already did it, just check it on my profile.

**Comment by tarasfrompir** [comment]
Did not start with apache 2.4 - just crash...

**Comment by SlardarZhang** [comment]
> Did not start with apache 2.4 - just crash...

PHP version should be 7.4.2, then you should put "opencv_world420.dll" in the same folder as php.exe< then put the "php_opencv.dll" in the extension folder, it should be run. If you want to determine its apache problem or installation problem, you can use the command line  "php.exe -m" to print the loaded modules list. If opencv in the list, then the problem should be caused by apache.
BTW: You MUST use php 7.4.2 TS version.
Good Luck

**Comment by tarasfrompir** [comment]
PHP Warning:  PHP Startup: Unable to load dynamic library 'opencv' (tried: C:\_majordomo\server\php\ext\opencv (╨Э╨╡ ╨╜╨░╨╣╨┤╨╡╨╜ ╤Г╨║╨░╨╖╨░╨╜╨╜╤Л╨╣ ╨╝╨╛╨┤╤Г╨╗╤М.), C:\_majordomo\server\php\ext\php_opencv.dll (%1 ╨╜╨╡ ╤П╨▓╨╗╤П╨╡╤В╤Б╤П ╨┐╤А╨╕╨╗╨╛╨╢╨╡╨╜╨╕╨╡╨╝ Win32.)) in Unknown on line 0

**Comment by SlardarZhang** [comment]
Looks like you need put opencv_world420.dll at the folder as php.exe. Otherwise, it may cause this problem. BTW, the x86 version has some problem, please use x64 version.

**Comment by tarasfrompir** [comment]
I try all version - php and opencv. I put files -php_opencv.dll and opencv_world420.dll - at the folder php and folder php/ext/ . 

But still getting the same error

**Comment by tarasfrompir** [comment]
Describe your compilation process on Windows

**Comment by SlardarZhang** [comment]
https://windows.php.net/downloads/releases/php-devel-pack-7.4.2-Win32-vc15-x64.zip
Download this dev-pack, use the phpize.bat to compile it.
I uploaded the Tester, you can test the opencv_world420.dll by this tool.
I think this problem caused by opencv_world420.dll.

**Comment by SlardarZhang** [comment]
> Describe your compilation process on Windows

https://windows.php.net/downloads/releases/php-devel-pack-7.4.2-Win32-vc15-x64.zip
Download this dev-pack, use the phpize.bat to compile it.
I uploaded the Tester, you can test the opencv_world420.dll by this tool.
I think this problem caused by opencv_world420.dll.

**Comment by SlardarZhang** [comment]
> Describe your compilation process on Windows


> I try all version - php and opencv. I put files -php_opencv.dll and opencv_world420.dll - at the folder php and folder php/ext/ .
> 
> But still getting the same error

Did the problem solve?

**Comment by tarasfrompir** [comment]
sorry - have not time to try...
Mi work is mi house...

**Comment by SlardarZhang** [comment]
> sorry - have not time to try...
> Mi work is mi house...

Okay, no worries.

**Comment by SlardarZhang** [comment]
> Thanks for your improvements dear SlardarZhang. But the test script doesn't work.
> The library has loaded, but no one of the functions work. Including a test script. The file path was specified explicit. What could be the problem ?
> 
> ```
> //Obama.php
> echo "ЕУІЕ" ;
> use function CV\{ imread, imshow, waitkey, namedWindow};
> 
>  if ($im = imread("http://smarthome.glory.pw/Obama.png")) {
>  echo "LOADED" ;
>  } else {
>  echo "NOLOADED";
>  }
>  namedWindow('This is Obama id card',WINDOW_FULLSCREEN);//create window
> imshow('This is Obama id card',$im);//show image on window
> 
> waitkey(0);
> ```
> 
> PHP 7.4
> ![image](https://user-images.githubusercontent.com/3198716/93341728-1fe1c180-f837-11ea-95c7-f8e28fdae370.png)

I will push a new version for PHP 7.4.10, please wait for a couple of weeks.

**Comment by tarasfrompir** [comment]
"I will push a new version for PHP 7.4.10, please wait for a couple of weeks.

"
Some one change ?

**Comment by SlardarZhang** [comment]
> "I will push a new version for PHP 7.4.10, please wait for a couple of weeks.
> 
> "
> Some one change ?

Yes, the php_opencv.dll have problem to call opencv_world420, BTW, I will compile OpenCV4.4 for the next version

**Comment by tarasfrompir** [comment]
Thanks i am wayting

**Comment by tarasfrompir** [comment]
> > "I will push a new version for PHP 7.4.10, please wait for a couple of weeks.
> > "
> > Some one change ?
> 
> Yes, the php_opencv.dll have problem to call opencv_world420, BTW, I will compile OpenCV4.4 for the next version

Hi SlardarZhang... How affairs with next version ?

**Comment by SlardarZhang** [comment]
I am waiting for PHP 8.0 release, currently, MS published a piece of news, they will not support PHP8.0, what do you think of it?

**Comment by tarasfrompir** [comment]
I am cry - wheare my PHPopencv - i dont lacky human......

**Comment by SlardarZhang** [comment]
> I am cry - wheare my PHPopencv - i dont lacky human......

Because Microsoft will not support PHP 8.0, do you think we can create a repository of PHP 8.0 for Windows? BTW, I am thinking about the next version of OpenCV for PHP library, I should build 7.4.12 or 8.0, could you give me some suggestion? thanks a lot.

**Comment by tarasfrompir** [comment]
I think - OpenCV for PHP 7.4.12.
Windows - is all my life

**Comment by SlardarZhang** [comment]
> I think - OpenCV for PHP 7.4.12.
> Windows - is all my life

The new version is released, you can download it now, this version I build a smart installer, it may help you to install it easier.
https://github.com/SlardarZhang/php-opencv/releases/tag/4.5.0%7C1.0.2

**Comment by tarasfrompir** [comment]
Very COOL - i try in mondey... Thanks

**Comment by SlardarZhang** [comment]
> Very COOL - i try in mondey... Thanks
Use 1.0.3 please, 1.0.2 have bugs.
https://github.com/SlardarZhang/php-opencv/releases/tag/4.5.0%7C1.0.3

**Comment by SlardarZhang** [comment]
> Very COOL - i try in mondey... Thanks

Use 1.0.3 please, 1.0.2 has a bug.
https://github.com/SlardarZhang/php-opencv/releases/tag/4.5.0%7C1.0.3

**Comment by tarasfrompir** [comment]
Thank you very much. We are waiting for version 8 PHP

**Comment by SlardarZhang** [comment]
> Thank you very much. We are waiting for version 8 PHP

PHP 8 compile have lots of problems, may need more time

**Comment by tarasfrompir** [comment]
I see you try do that. so we are waiting...
